app/knights/mordred.py

```diff
@@ -0,0 +1,23 @@
+class MordRed:
+
+    def __init__(self, pers_name: dict) -> None:
+        self.pers_name = pers_name
+        self.name = pers_name["name"]
+        self.power = pers_name["power"]
+        self.hp = pers_name["hp"]
+        self.armour = pers_name["armour"]
+        self.weapon = pers_name["weapon"]
+        self.potion = pers_name["potion"]
+
+    def mordred_armour(self) -> list | None:
+        if not self.armour:
+            return None
+        return self.armour
+
+    def mordred_weapon(self) -> dict:
+        return self.weapon
+
+    def mordred_potion(self) -> dict | None:
+        if self.potion is None:
+            return None
+        return self.potion["effect"]
```

Comment:
don't repeat yourself 

create separate method `knight_potion` and use it on all knights

Reply:
resolve it. don't create separate class for each knight